### PR TITLE
chore: improve header type

### PR DESCRIPTION
**Pull request by septs**
reduce the same code (`map[string]Listable[string]` to `http.Header`)

### CODE REVIEW

File: option/types.go [hunk 1]
```diff
@@ -234,3 +235,13 @@ func DNSQueryTypeToString(queryType uint16) string {
 	}
 	return F.ToString(queryType)
 }
+
+type OutboundHeader map[string]Listable[string]
+
+func (h OutboundHeader) HTTPHeader() http.Header {
+	header := make(http.Header)
+	for name, values := range h {
+		header[http.CanonicalHeaderKey(name)] = values
```
Comment (septs) [inline option/types.go:244]: https://pkg.go.dev/net/http#CanonicalHeaderKey

**Comment by nekohasekai** [comment]
I'm afraid they are not the same, and v2ray allows such mixing. You cannot remove a feature and cause configuration breaking unless there is good reason.

**Comment by septs** [comment]
https://github.com/SagerNet/sing-box/blob/aa3255f7634f10f43f91ef075bcf7b6021073052/option/types.go#L239

I didn't break compatibility

**Comment by nekohasekai** [comment]
https://www.v2fly.org/config/transport/tcp.html#httprequestobject

**Comment by septs** [comment]
https://github.com/SagerNet/sing-box/blob/aa3255f7634f10f43f91ef075bcf7b6021073052/option/types.go#L239

I didn't remove the `Listable`

**Pull request merged**